pulp_smash/constants.py

```diff
@@ -343,11 +344,74 @@
 .. _Repository APIs:
     https://docs.pulpproject.org/en/latest/dev-guide/integration/rest-api/repo/index.html
 """
-
-RPM = 'bear-4.1-1.noarch.rpm'
+RPM_DATA = MappingProxyType({
+    'name': 'bear',
+    'epoch': '0',
+    'version': '4.1',
+    'release': '1',
+    'arch': 'noarch',
+    'vendor': None,
+    'metadata': {
+        'release': '1',
+        'license': 'GPLv2',
+        'description': 'A dummy package of bear',
+        'files': {'dir': [], 'file': ['/tmp/bear.txt']},
+        'group': 'Internet/Applications',
+        'size': '42',
+        'sourcerpm': 'bear-4.1-1.src.rpm',
+        'summary': 'A dummy package of bear',
+    },
+})
+"""Metadata for an RPM with an associated erratum.
+
+The metadata tags that may be present in an RPM may be printed with:
+
+.. code-block:: sh
+
+    rpm --querytags
+
+Metadata for an RPM can be printed with a command like the following:
+
+.. code-block:: sh
+
+    for tag in name epoch version release arch vendor; do
+        echo "$(rpm -qp bear-4.1-1.noarch.rpm --qf "%{$tag}")"
+    done
+
+"""
+RPM = '{}-{}{}-{}.{}.rpm'.format(
+    RPM_DATA['name'],
+    RPM_DATA['epoch'] + '!' if RPM_DATA['epoch'] != '0' else '',
+    RPM_DATA['version'],
+    RPM_DATA['release'],
+    RPM_DATA['arch'],)
```

Comment:
The reason for adding a trailing comma to the end of a list/tuple/etc is that:

* It makes diffs simpler.
* It makes changes less error-prone.

Consider this code:

```python
t = (
    1,
    2,
    3
)
```

If I want to add "4" to the list, I might make this diff

```diff
4a5
>     4
```

But this syntax is wrong, because I've forgotten to add a comma after the "3". If I then add a comma after the "3", then the diff looks like this:

```diff
4c4,5
<     3
---
>     3,
>     4
```

...which works, but is verbose. On the other hand, if I start with:

```python
t = (
    1,
    2,
    3,
)
```

Then my diff can be this:

```diff
4a5
>     4,
```

This diff is both concise *and* correct.

...and now you know.